addons/social_share/controllers/main.py

```diff
@@ -0,0 +1,104 @@
+import base64
+
+from odoo.http import Controller, content_disposition, request, route
+
+from .user_agents import NETWORK_TO_AGENT
+from ..utils.image_utils import scale_image
+
+class SocialShareController(Controller):
+
+    @route(['/snshare/cards/<int:campaign_id>/card.png', '/snshare/cards/<int:campaign_id>/<string:uuid>/card.png'], type='http', auth='public', sitemap=False, website=True)
+    def snshare_campaign_image(self, campaign_id, uuid=None, small=False):
+        campaign_sudo = request.env['snshare.campaign'].sudo().browse(campaign_id).exists()
+        url_sudo = request.env['snshare.url'].sudo()
+        target = None
+        if campaign_sudo.model_id:
+            if not uuid:
+                return request.not_found()
+            url_sudo = request.env['snshare.url'].sudo().search([('campaign_id', '=', campaign_sudo.id), ('uuid', '=', uuid)])
+            target = request.env[campaign_sudo.model_id.model].sudo().browse(url_sudo.res_id).exists()
+
+        crawler = self._get_crawler_name(request)
+        if crawler and not url_sudo.is_shared:
+            request.env['bus.bus']._sendone(f'snshare_url_target-{uuid}', 'snshare/share_url_target', {
+                'message': campaign_sudo.thanks_message,
+                'reward_url': campaign_sudo.thanks_redirection,
+            })
+            url_sudo.is_shared = True
+
+        image_bytes = base64.b64decode(url_sudo._get_image()) if url_sudo else campaign_sudo._get_images_bytes(record=target)[0]
+        image_bytes = image_bytes if not small else scale_image(image_bytes, 0.5)
+        return request.make_response(image_bytes, [
+            ('Content-Type', ' image/png'),
+            ('Content-Length', len(image_bytes)),
+            ('Content-Disposition', content_disposition('card.png')),
+        ])
+
+    @route(['/snshare/campaign/<int:campaign_id>', '/snshare/campaign/<int:campaign_id>/<string:uuid>'], type='http', auth='public', sitemap=False, website=True)
+    def snshare_campaign_visitor(self, campaign_id, uuid=None):
```

Comment:
Can't we simplify to something like `/cards/<int:campaign_id>/<string:uuid>/preview` ? Would use the preview part as it helps differentiating from the other route

Reply:
If they don't, tant pis. Purpose is to have clean and understandable routes, using a model / id / method approach.

Reply:
Hm, not really tanpis as it's an allow list so if they don't support it they probably won't follow the link to get the open-graph metadata

I'll see if it's an issue for the supported website